code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -565,18 +602,58 @@
 				else
 					message = "sadly can't find anybody to give daps to, and daps [get_visible_gender() == MALE ? "himself" : get_visible_gender() == FEMALE ? "herself" : "themselves"]. Shameful."
 
-		if ("scream")
-			if (miming)
+		if("slap", "slaps")
+			m_type = 1
+			if(!restrained())
+				var/M = null
+				if(param)
+					for(var/mob/A in view(1, null))

```

Comment:
I can fix that. There should really be a cooldown for emotes, though, as on Paradise.